demo-circuit/src/naive_threshold_sig_w_key_rotation/tests/utils.rs

```diff
@@ -93,3 +93,34 @@ pub(crate) fn updated_key_msg(pk: FieldBasedSchnorrPk<G2Projective>) -> FieldEle
     });
     h.finalize().unwrap()
 }
+
+pub(crate) fn rotate_key(
+    signing_key_sk: &SchnorrSk,
+    signing_key_pk: &FieldBasedSchnorrPk<G2Projective>,
+    master_key_sk: &SchnorrSk,
+    master_key_pk: &FieldBasedSchnorrPk<G2Projective>,
+    updated_sk_signatures: &mut FieldBasedSchnorrSignature<FieldElement, G2Projective>,
+    updated_mk_signatures: &mut FieldBasedSchnorrSignature<FieldElement, G2Projective>,
+    updated_signing_sks: &mut SchnorrSk,
```

Comment:
I would use an Option here to deal with cases when the caller does not need to update the signing secret key; in this way, the caller can avoid allocating a dummy secret key just to call this function. Alternatively, the updated key pair can be returned to the caller rather than being set inside the function

Reply:
Done